smarty/sampler.py

```diff
@@ -97,6 +98,18 @@ def __init__(self, molecules, basetypes_filename, initialtypes_filename, decorat
         # Save bond list to use throughout
         bondset = [("-","singly"), ("=", "doubly"), ("#","triply"), (":", "aromatic")]
 
+        used_basetypes = list()
+        self.atomtypes_with_no_matches = set()
+        # Check all SMART strings that are used as a base type
+        for (smarts, atom_type) in self.basetypes:
+            check_flag = self.smarts_matches(smarts)
+            if check_flag == True:
+                used_basetypes.append( ( smarts, atom_type) )
+            else:
+                self.atomtypes_with_no_matches.add( smarts )
```

Comment:
Checking if something `== True` is redundant. You can just call:
```
if self.smarts_matches(smarts):
    used_basetypes.append( ( smarts, atom_type) )
else:
    self.atomtypes_with_no_matches.add( smarts )
```

Or if you think it is more clear to define check_flag then you could just say `if check_flag:` on line 106.